tools/manifest/vcs.py

```diff
@@ -224,17 +224,61 @@ def dump(self):
         super(MtimeCache, self).dump()
 
 
-class GitIgnoreCache(CacheFile, GitIgnoreCacheType):
-    file_name = "gitignore2.json"
+class GitIgnoreCache(GitIgnoreCacheType):
+    file_name = "gitignore"
 
-    def check_valid(self, data):
-        # type: (Dict[Any, Any]) -> Dict[Any, Any]
-        ignore_path = os.path.join(self.tests_root, ".gitignore")
-        mtime = os.path.getmtime(ignore_path)
-        if data.get(u"/gitignore_file") != [ignore_path, mtime]:
-            self.modified = True
+    def __init__(self, cache_root, tests_root, rebuild=False):
+        # type: (Text, Text, bool) -> None
+        self.tests_root = tests_root
+        if not os.path.exists(cache_root):
+            os.makedirs(cache_root)
+
+        self.path = os.path.join(cache_root, self.file_name)
+        self.ignore_path = os.path.join(self.tests_root, ".gitignore")
+        self.ignore_path_bytes = self.ignore_path.encode("utf8")
+
+        self.modified = False
+        self.data = self.load(rebuild)
+
+    def get_cache_key(self):
+        # type: () -> Tuple[bytes, int]
+        mtime = int(os.path.getmtime(self.ignore_path) * 1000)
+        return (self.ignore_path_bytes, mtime)
+
+    def dump(self):
+        # type: () -> None
+        if not self.modified:
+            return
+        with open(self.path, 'wb') as f:
+            f.write(b"%s\0%d\n" % self.get_cache_key())
```

Comment:
One regret I have (and what led to `gitignore2.json`) is not having a cache-version in the cache key. Can we add one, at least here?